### PR TITLE
Billing Snapshot Config APIs introduced in Usage Reports

### PR DESCRIPTION
**Pull request by manukm-ibm**
@kavya498 @hkantare Could you please review this PR

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccIBMBillingReportSnapshotBasic'

=== RUN   TestAccIBMBillingReportSnapshotBasic
--- PASS: TestAccIBMBillingReportSnapshotBasic (60.96s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/usagereports	62.503s

=== RUN   TestAccIBMBillingSnapshotListDataSourceBasic
--- PASS: TestAccIBMBillingSnapshotListDataSourceBasic (29.37s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/usagereports	30.801s
```

### CODE REVIEW

File: ibm/service/usagereports/resource_ibm_billing_report_snapshot.go [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+func ResourceIBMBillingReportSnapshot() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMBillingReportSnapshotCreate,
+		ReadContext:   resourceIBMBillingReportSnapshotRead,
+		UpdateContext: resourceIBMBillingReportSnapshotUpdate,
+		DeleteContext: resourceIBMBillingReportSnapshotDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": &schema.Schema{
```
Comment (kavya498) [inline ibm/service/usagereports/resource_ibm_billing_report_snapshot.go:30]: Acccount id can be taken internally in the code.
Do we really need user to pass accountid?
Same applies to datasource as well.

Comment (manukm-ibm) [reply ibm/service/usagereports/resource_ibm_billing_report_snapshot.go]: Incorporated the change to pick account id internally.

File: ibm/service/usagereports/data_source_ibm_billing_snapshot_list.go [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+	getReportsSnapshotOptions.SetAccountID(userDetails.UserAccount)
+	getReportsSnapshotOptions.SetMonth(d.Get("month").(string))
+	if _, ok := d.GetOk("date_from"); ok {
+		getReportsSnapshotOptions.SetDateFrom(int64(d.Get("date_from").(int)))
+	}
+	if _, ok := d.GetOk("date_to"); ok {
+		getReportsSnapshotOptions.SetDateTo(int64(d.Get("date_to").(int)))
+	}
+
+	snapshotList, response, err := usageReportsClient.GetReportsSnapshotWithContext(context, getReportsSnapshotOptions)
```
Comment (kavya498) [inline ibm/service/usagereports/data_source_ibm_billing_snapshot_list.go:232]: Pagination is not handled in this datasource..
Kindly handle pagination and remove first and last arguments from schema.

After removing these arguments from schema remove from docs as well.

File: ibm/service/usagereports/resource_ibm_billing_report_snapshot.go [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+			"history": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of previous versions of the snapshot configurations.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"start_time": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Description: "Timestamp in milliseconds when the snapshot configuration was created.",
```
Comment (kavya498) [inline ibm/service/usagereports/resource_ibm_billing_report_snapshot.go:109]: All child arguments of a computed field should be computed. It cannot be Optional.

**Review by kavya498 — CHANGES_REQUESTED** [review]

File: ibm/provider/provider.go [hunk 4]
```diff
@@ -706,8 +710,6 @@ func Provider() *schema.Provider {
 			"ibm_satellite_link":                                satellite.DataSourceIBMSatelliteLink(),
 			"ibm_satellite_endpoint":                            satellite.DataSourceIBMSatelliteEndpoint(),
 			"ibm_satellite_cluster_worker_pool_zone_attachment": satellite.DataSourceIBMSatelliteClusterWorkerPoolAttachment(),
-			"ibm_satellite_storage_configuration":               satellite.DataSourceIBMSatelliteStorageConfiguration(),
```
[the remark this answers is not in the document]
Comment (manukm-ibm) [reply ibm/provider/provider.go]: Added them back. Might be missed from one of the merge commits.

File: ibm/service/usagereports/data_source_ibm_billing_snapshot_list.go (hunk 2 above)
Comment (manukm-ibm) [reply ibm/service/usagereports/data_source_ibm_billing_snapshot_list.go]: Added pagination support.

Comment (manukm-ibm) [reply ibm/service/usagereports/data_source_ibm_billing_snapshot_list.go]: Removed the first and next arguments from the schema as well.

File: ibm/service/usagereports/resource_ibm_billing_report_snapshot.go (hunk 3 above)
Comment (manukm-ibm) [reply ibm/service/usagereports/resource_ibm_billing_report_snapshot.go:109]: Done.

**Review by kavya498 — APPROVED** [review]

**Pull request merged**